internal/cache/file/cache_handle.go

```diff
@@ -119,34 +123,48 @@ func (fch *CacheHandle) Read(ctx context.Context, object *gcs.MinObject, bucket
 		requiredOffset = objSize
 	}
 
+	// If downloadForRandomRead is false and readType is random, download will not be initiated.
+	if !downloadForRandomRead && !fch.IsSequential(offset) {
```

Comment:
can we save value of fch.IsSequential(offset) in a variable above and use that here instead of making another call ?